app/controllers/affiliate_requests/onboarding_form_controller.rb

```diff
@@ -11,19 +11,36 @@ def update
     user_product_params = permitted_params[:products].filter { |product| user_product_external_ids.include?(product[:id].to_i) }
 
     if disabling_all_products_while_having_pending_requests?(user_product_params)
-      return render json: { success: false, error: "You need to have at least one product enabled since there are some pending affiliate requests" }
+      message = "You need to have at least one product enabled since there are some pending affiliate requests"
+      if request.inertia?
+        return redirect_back(fallback_location: root_path, alert: message)
```

Comment:
You should probably always redirect to the affiliates onboarding page, unless this is not the only place where the affiliates onboarding form is shown